### PR TITLE
stats: set a min count for auto analyze

### PR DESCRIPTION
**Pull request by hanfei1991**
For small table, we need not auto analyze.

### CODE REVIEW

File: statistics/update_test.go [hunk 1]
```diff
@@ -269,6 +270,11 @@ func (s *testStatsUpdateSuite) TestAutoUpdate(c *C) {
 	testKit.MustExec("use test")
 	testKit.MustExec("create table t (a int)")
 
+	statistics.AutoAnalyzeMinCnt = 0
+	defer func() {
+		statistics.AutoAnalyzeMinCnt = 1000
```
Comment (winoros) [inline statistics/update_test.go]: 1000 is too small?

Comment (hanfei1991) [reply statistics/update_test.go]: I think 1000 is a conservative value. If it works well in production, we can scale it up.

**Comment by shenli** [comment]
LGTM

**Pull request merged**